### PR TITLE
adding create_dataentity

### PR DESCRIPTION
**Pull request by pariosur**
Adding: 
- Create Data Entity method

### CODE REVIEW

File: src/metal_sdk/metal.py [hunk 1]
```diff
@@ -308,6 +308,45 @@ def update_datasource(self, id: str, payload: dict):
...
+        if not os.path.exists(file_path):
+            raise ValueError(f"File '{file_path}' not found.")
+
+        file_size = os.path.getsize(file_path)
+        filename = os.path.basename(file_path)
+        file_type, _ = mimetypes.guess_type(file_path)
+
+        # Restrict to CSV and PDF files only.
+        supported_types = ['application/pdf', 'text/csv']
+        if file_type not in supported_types:
```
Comment (softboyjimbo) [inline src/metal_sdk/metal.py:337]: we accept other file types now - docx, xlsx, etc. the api **should** restrict this as well so may be easier to let the server do its thing. what do you think? cc @Czechh

Comment (pariosur) [reply src/metal_sdk/metal.py:337]: awesome - will modify

Comment (Czechh) [reply src/metal_sdk/metal.py:337]: yeahh good call!

**Comment by pariosur** [comment]
@Czechh @softboyjimbo added tests for the new functions

**Review by Czechh — APPROVED** [review]

**Comment by pariosur** [comment]
@Czechh @softboyjimbo  added the create_index method too, please lmk if it looks ok

File: src/metal_sdk/metal.py [hunk 2]
```diff
@@ -308,21 +308,46 @@ def update_datasource(self, id: str, payload: dict):
         res = self.fetch("put", url, payload)
         return res
 
-    def get_dataentity(self, id: str):
-        if id is None:
-            raise TypeError("dataentity_id required")
+    def __create_dataentity_resource(self, datasource, filename, file_size):
```
Comment (Czechh) [inline src/metal_sdk/metal.py:311]: Nit Here I think `datasource` is correct to be one word, but `dataentity` should be `data_entity`

File: src/metal_sdk/metal.py [hunk 3]
```diff
@@ -308,21 +308,46 @@ def update_datasource(self, id: str, payload: dict):
...
+        }
+        headers = {'x-metal-file-size': str(file_size)}
 
-        url = f"/v1/data-entities/{id}"
-        res = self.fetch("get", url, None)
-        return res
+        return self.fetch("post", url, payload, headers=headers)
 
-    def delete_dataentity(self, id: str):
+    def create_dataentity(self, datasource, file_path):
```
Comment (Czechh) [inline src/metal_sdk/metal.py:323]: same here `create_data_entity`

File: src/metal_sdk/metal.py [hunk 4]
```diff
@@ -308,21 +308,46 @@ def update_datasource(self, id: str, payload: dict):
...
+        if not resource or 'data' not in resource:
+            logger.error("Failed to create a data entity resource.")
+            return None
+
+        # Upload the file to the returned URL.
+        self.__upload_file_to_url(resource['data']['url'], file_path, file_type, file_size)
+
+        return resource
+
+    def get_dataentity(self, id: str):
```
Comment (Czechh) [inline src/metal_sdk/metal.py:345]: `get_data_entity`

File: src/metal_sdk/metal.py [hunk 5]
```diff
@@ -340,3 +365,17 @@ def get_all_dataentities(self, datasource_id: str, limit=None, page=None):
 
         res = self.fetch("get", url, None, params)
         return res
+
+    def delete_dataentity(self, id: str):
```
Comment (Czechh) [inline src/metal_sdk/metal.py:369]: `delete_data_entity`

**Review by Czechh — COMMENTED** [review]
Just one comment

**Review by Czechh — CHANGES_REQUESTED** [review]
Some changes needed

File: src/metal_sdk/metal.py (hunk 2 above)
Comment (pariosur) [reply src/metal_sdk/metal.py]: perfect! fixed

**Comment by pariosur** [comment]
@Czechh fixed! also done for data_entities

**Pull request merged**